test/server/cards/01_SOR/units/BobaFettDisintegrator.spec.js

```diff
@@ -0,0 +1,119 @@
+describe('Boba Fett Disintegrator', function() {
+    integration(function() {
+        describe('Boba Fett Disintegrator\'s ability', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['boba-fett#disintegrator'],
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'consular-security-force'],
+                        leader: 'luke-skywalker#faithful-friend',
+                        hand: ['lom-pyke#dealer-in-truths'],
+                    }
+                });
+            });
+
+            it('should have no effect when attacking a ready unit or a exhausted unit played this turn.', function () {
+                // Case 1 attacking a ready card
+                expect(this.player1).toBeActivePlayer();
+                this.player1.clickCard(this.bobaFett);
+                this.player1.clickCard(this.wampa);
+
+                // check board state
+                expect(this.wampa.damage).toBe(3);
+                expect(this.bobaFett.damage).toBe(4);
+
+                // reset board state
+                this.bobaFett.damage = 0;
+                this.bobaFett.exhausted = false;
+                this.wampa.damage = 0;
+
+                // Case 2 attacking a card played this turn
+                expect(this.player2).toBeActivePlayer();
+                this.player2.clickCard(this.lomPyke);
+                this.player1.clickCard(this.bobaFett);
+                this.player1.clickCard(this.lomPyke);
+
+                // check board state
+                expect(this.player2).toBeActivePlayer();
+                expect(this.lomPyke.damage).toBe(3);
+                expect(this.bobaFett.damage).toBe(4);
+            });
+
+            it('Should activate when attacking a exhausted unit not played in this phase.', function() {
```

Comment:
Two things:

1. We can stitch all of these test cases together into one I think
2. For the test(s) where the Boba effect is expected to do damage, instead of targeting a unit that was put in place by the test (since that works a little weird) let's play unit from hand, then use `this.moveToNextActionPhase()`

Reply:
When it comes to this comment i'm not quite sure what you mean. Do you mean to create 1 it() function with all the cases or multiple it() functions with one describe?. I'll fix number 2 when nr 1 is clarified.

Reply:
Oh yeah sorry - it's the firs tthing you said, create 1 `it()` with multiple cases. It's because each `it()` takes a very long time to run (compared to other test frameworks I've used). Take a look at the first test case for General Krell to see what I mean: https://github.com/SWU-Karabast/forceteki/blob/main/test/server/cards/01_SOR/units/GeneralKrellHeartlessTactician.spec.js

Reply:
No worries, i'll rewrite the test cases. Thanks for the explanation.